flink-tests/src/test/scala/org/apache/flink/api/scala/operators/SampleITCase.scala

```diff
@@ -0,0 +1,166 @@
+
+import java.util.{List => JavaList, Random}
+
+import org.apache.flink.api.scala._
+import org.apache.flink.api.scala.util.CollectionDataSets
+import org.apache.flink.test.util.MultipleProgramsTestBase.TestExecutionMode
+import org.apache.flink.test.util.{MultipleProgramsTestBase, TestBaseUtils}
+import org.junit.Assert._
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.{Before, After, Test}
+
+import scala.collection.JavaConverters._
+
+@RunWith(classOf[Parameterized])
+class SampleITCase(mode: TestExecutionMode) extends MultipleProgramsTestBase(mode) {
+  private val RNG: Random = new Random
+
+  private var result: JavaList[String] = null;
+
+  @Before
+  def initiate {
+    ExecutionEnvironment.getExecutionEnvironment.setParallelism(5)
+  }
+
+  @After
+  def after() = {
+    TestBaseUtils.containsResultAsText(result, getSourceStrings)
+  }
+
+  @Test
+  @throws(classOf[Exception])
+  def testSamplerWithFractionWithoutReplacement {
+    verifySamplerWithFractionWithoutReplacement(0d)
+    verifySamplerWithFractionWithoutReplacement(0.2d)
+    verifySamplerWithFractionWithoutReplacement(1.0d)
+  }
+
+  @Test
+  @throws(classOf[Exception])
+  def testSamplerWithFractionWithReplacement {
+    verifySamplerWithFractionWithReplacement(0d)
+    verifySamplerWithFractionWithReplacement(0.2d)
+    verifySamplerWithFractionWithReplacement(1.0d)
+    verifySamplerWithFractionWithReplacement(2.0d)
+  }
+
+  @Test
+  @throws(classOf[Exception])
+  def testSamplerWithSizeWithoutReplacement {
+    verifySamplerWithFixedSizeWithoutReplacement(0)
+    verifySamplerWithFixedSizeWithoutReplacement(2)
+    verifySamplerWithFixedSizeWithoutReplacement(21)
+  }
+
+  @Test
+  @throws(classOf[Exception])
+  def testSamplerWithSizeWithReplacement {
+    verifySamplerWithFixedSizeWithReplacement(0)
+    verifySamplerWithFixedSizeWithReplacement(2)
+    verifySamplerWithFixedSizeWithReplacement(21)
+  }
+
+  @throws(classOf[Exception])
+  private def verifySamplerWithFractionWithoutReplacement(fraction: Double) {
+    verifySamplerWithFractionWithoutReplacement(fraction, RNG.nextLong)
+  }
+
+  @throws(classOf[Exception])
+  private def verifySamplerWithFractionWithoutReplacement(fraction: Double, seed: Long) {
+    verifySamplerWithFraction(false, fraction, seed)
+  }
+
+  @throws(classOf[Exception])
+  private def verifySamplerWithFractionWithReplacement(fraction: Double) {
+    verifySamplerWithFractionWithReplacement(fraction, RNG.nextLong)
+  }
+
+  @throws(classOf[Exception])
+  private def verifySamplerWithFractionWithReplacement(fraction: Double, seed: Long) {
+    verifySamplerWithFraction(true, fraction, seed)
+  }
+
+  @throws(classOf[Exception])
+  private def verifySamplerWithFraction(withReplacement: Boolean, fraction: Double, seed: Long) {
+    val ds = getSourceDataSet()
+    val sampled = ds.sample(withReplacement, fraction, seed)
+    result = sampled.collect.asJava
```

Comment:
The validity of sample result is verified in after() method for each test. As the source data is very small, verify the fraction does not make much sense, so i didn't verify the fraction validity here, but it got verified in RandomSamplerTest in Sampler level by the way.